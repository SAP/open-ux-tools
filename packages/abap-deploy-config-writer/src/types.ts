export interface DeployConfigOptions {
    baseFile?: string; // e.g ui5.yaml
    deployFile?: string; // e.g ui5-deploy.yaml
}
