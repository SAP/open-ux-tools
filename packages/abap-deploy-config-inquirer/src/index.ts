import { ToolsLogger, type Logger } from '@sap-ux/logger';
import { initI18n } from './i18n';
import { PromptState } from './prompts/prompt-state';
import { getAbapDeployConfigQuestions } from './prompts';
import LoggerHelper from './logger-helper';
import type { InquirerAdapter } from '@sap-ux/inquirer-common';
import type { AbapDeployConfigAnswers, AbapDeployConfigPromptOptions } from './types';

/**
 * Get the inquirer prompts for abap deploy config.
 *
 * @param promptOptions - options that can control some of the prompt behavior. See {@link AbapDeployConfigPromptOptions} for details
 * @param logger - a logger compatible with the {@link Logger} interface
 * @returns the prompts used to provide input for abap deploy config generation and a reference to the answers object which will be populated with the user's responses once `inquirer.prompt` returns
 */
async function getPrompts(
    promptOptions?: AbapDeployConfigPromptOptions,
    logger?: Logger
): Promise<{ prompts: any[]; answers: Partial<AbapDeployConfigAnswers> }> {
    await initI18n();
    LoggerHelper.logger = logger ?? new ToolsLogger({ logPrefix: '@sap-ux/abap-deploy-config-inquirer' });

    return {
        prompts: await getAbapDeployConfigQuestions(promptOptions),
        // Return reference to derived answers object that will be populated with user responses (after prompting is complete)
        answers: PromptState.abapDeployConfig
    };
}

/**
 * Prompt for abap deploy config writer inputs.
 *
 * @param adapter - optionally provide references to a calling inquirer instance, this supports integration to Yeoman generators, for example
 * @param promptOptions - options that can control some of the prompt behavior. See {@link AbapDeployConfigPromptOptions} for details
 * @param logger - a logger compatible with the {@link Logger} interface
 * @returns the prompt answers
 */
async function prompt(
    adapter: InquirerAdapter,
    promptOptions?: AbapDeployConfigPromptOptions,
    logger?: Logger
): Promise<AbapDeployConfigAnswers> {
    const abapDeployConfigPrompts = (await getPrompts(promptOptions, logger)).prompts;
    const answers = await adapter.prompt<AbapDeployConfigAnswers>(abapDeployConfigPrompts);

    // Add dervied service answers to the answers object
    Object.assign(answers, PromptState.abapDeployConfig);

    return answers;
}

export { getPrompts, prompt, type InquirerAdapter, type AbapDeployConfigAnswers, type AbapDeployConfigPromptOptions };
