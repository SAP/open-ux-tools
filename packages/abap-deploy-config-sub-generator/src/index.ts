export { getAbapQuestions } from './app/questions';
export { indexHtmlExists, showOverwriteQuestion } from './utils';
export { AbapDeployConfigOptions, DeployProjectType } from './app/types';
