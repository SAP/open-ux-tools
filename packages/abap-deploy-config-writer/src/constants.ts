export const BUILD_SCRIPT = 'npm run build';
export const DEPLOY_SCRIPT = 'fiori deploy';
export const UNDEPLOY_SCRIPT = 'fiori undeploy';
export const RIMRAF_VERSION = '^5.0.5';
export const RIMRAF = 'rimraf';
export const UI5_TASK_FLATTEN_LIB = 'ui5-task-flatten-library';
export const UI5_TASK_FLATTEN_LIB_VERSION = '^3.1.0';
export const UI5_REPO_TEXT_FILES = '.Ui5RepositoryTextFiles';
export const UI5_REPO_IGNORE = '.Ui5RepositoryIgnore';
export const UI5_CLI_LIB = '@ui5/cli';
export const UI5_CLI_MIN_VERSION = '3.0.0';
