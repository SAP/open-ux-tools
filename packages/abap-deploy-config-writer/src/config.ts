import { isAppStudio } from '@sap-ux/btp-utils';
import { UI5Config } from '@sap-ux/ui5-config';
import { UI5_TASK_FLATTEN_LIB } from './constants';
import type { Editor } from 'mem-fs-editor';
import type { AbapDeployConfig, AbapTarget, FioriToolsProxyConfig } from '@sap-ux/ui5-config';

/**
 * Updates the base config with the required custom tasks.
 *
 * @param basePath - the base path
 * @param baseConfig - the base config
 * @param fs - the memfs editor instance
 */
export function updateBaseConfig(basePath: string, baseConfig: UI5Config, fs: Editor) {
    if (baseConfig.getType() === 'library') {
        const customTask = {
            name: UI5_TASK_FLATTEN_LIB,
            afterTask: 'generateResourcesJson'
        };
        baseConfig.addCustomTasks([customTask]);
        fs.write(basePath, baseConfig.toString());
        baseConfig.removeConfig('builder');
    }
}

/**
 * Returns the deployment config.
 *
 * @param config - the deploy config
 * @param baseConfig - the base config
 * @returns the deploy config
 */
export async function getDeployConfig(config: AbapDeployConfig, baseConfig: UI5Config): Promise<UI5Config> {
    const target: Partial<AbapTarget> = {};

    if (config.target.destination !== undefined) {
        target.destination = config.target.destination;
    }
    if (config.target.url !== undefined) {
        target.url = config.target.url;
    }
    if (config.target.client) {
        target.client = config.target.client;
    }
    if (config.target.scp) {
        target.scp = true;
    }

    if (!isAppStudio()) {
        const middleware = baseConfig.findCustomMiddleware<FioriToolsProxyConfig>('fiori-tools-proxy');
        if (middleware?.configuration?.backend?.[0].authenticationType === 'reentranceTicket') {
            target.authenticationType = 'reentranceTicket';
        }
    }
    const baseUi5Doc = baseConfig.removeConfig('server');
    const ui5DeployConfig = await UI5Config.newInstance(baseUi5Doc.toString());

    ui5DeployConfig.addComment({
        comment: ' yaml-language-server: $schema=https://sap.github.io/ui5-tooling/schema/ui5.yaml.json',
        location: 'beginning'
    });

    ui5DeployConfig.addAbapDeployTask(target as unknown as AbapTarget, config.app, true, ['/test/'], config.index);

    return ui5DeployConfig;
}
