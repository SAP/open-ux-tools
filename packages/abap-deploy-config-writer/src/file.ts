import { join } from 'path';
import fg from 'fast-glob';
import { platform } from 'os';
import { FileName, type Package, getWebappPath } from '@sap-ux/project-access';
import { UI5_CLI_LIB, UI5_CLI_MIN_VERSION, UI5_REPO_TEXT_FILES } from './constants';
import { coerce, satisfies } from 'semver';
import type { Editor } from 'mem-fs-editor';

/**
 * Adds a new UI5 dependency to the package.json.
 * No longer required for `@ui5/cli` v3 and above.
 *
 * @param fs - the memfs editor instance
 * @param basePath - the base path
 * @param depName - the dependency name
 */
export function addUi5Dependency(fs: Editor, basePath: string, depName: string): void {
    const filePath = join(basePath, FileName.Package);
    const packageJson = (fs.readJSON(filePath) ?? {}) as Package;

    const ui5CliVersion = coerce(packageJson?.devDependencies?.[UI5_CLI_LIB]);
    if (ui5CliVersion && satisfies(ui5CliVersion, `>=${UI5_CLI_MIN_VERSION}`)) {
        // https://sap.github.io/ui5-tooling/v3/updates/migrate-v3/#changes-to-dependency-configuration
        return;
    }

    packageJson.ui5 = packageJson.ui5 ?? {};
    packageJson.ui5.dependencies = packageJson.ui5.dependencies ?? [];

    if (!packageJson.ui5.dependencies.includes(depName)) {
        packageJson.ui5.dependencies.push(depName);
    }
    fs.writeJSON(filePath, packageJson);
}

/**
 * Writes the UI5 repository file.
 *
 * @param fs - the memfs editor instance
 * @param basePath - the base path
 * @param ui5RepositoryFile - the UI5 repository file
 * @param addContent - the content to be added
 */
export const writeUi5RepositoryFile = (
    fs: Editor,
    basePath: string,
    ui5RepositoryFile: string,
    addContent: string
): void => {
    const filePath = join(basePath, ui5RepositoryFile);
    let content: string;
    if (fs.exists(filePath)) {
        content = fs.read(filePath);
        if (!content.includes(addContent)) {
            content = `${content}\n${addContent}`;
        }
    } else {
        content = addContent;
    }
    fs.write(filePath, content);
};

/**
 * Writes the UI5 repository files if typescript files are found.
 *
 * @param fs - the memfs editor instance
 * @param basePath - the base path
 */
export const writeUi5RepositoryFiles = async (fs: Editor, basePath: string): Promise<void> => {
    const webappPath = await getWebappPath(basePath);
    const typescriptPattern = join(webappPath, '/**/*.ts');
    const normalisedPath = platform() === 'win32' ? typescriptPattern.replace(/\\/g, '/') : typescriptPattern;
    const typeScriptFilesPaths: string[] = await fg(normalisedPath);
    if (typeScriptFilesPaths?.length > 0) {
        writeUi5RepositoryFile(fs, webappPath, UI5_REPO_TEXT_FILES, '^.*.ts$');
    }
};
