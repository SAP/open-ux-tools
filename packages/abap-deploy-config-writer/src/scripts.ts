import type { Editor } from 'mem-fs-editor';

import { addPackageDevDependency, updatePackageScript } from '@sap-ux/project-access';

import type { DeployConfigOptions } from './types';
import { BUILD_SCRIPT, DEPLOY_SCRIPT, RIMRAF, RIMRAF_VERSION, UNDEPLOY_SCRIPT } from './constants';

/**
 * Updates the scripts in the package.json file with the provided scripts object.
 *
 * @param {string} basePath - The path to the base directory.
 * @param {string} deployConfigFile - The path to the deploy config file.
 * @param {Editor} fs - The file system editor.
 * @param {DeployConfigOptions} options - The deploy config options.
 */
export async function updateScripts(
    basePath: string,
    deployConfigFile: string,
    fs: Editor,
    options?: DeployConfigOptions
): Promise<void> {
    const buildPrefix = options?.addBuildToUndeployScript ?? true ? `${BUILD_SCRIPT} && ` : '';

    // deploy script
    const deployScript = `${BUILD_SCRIPT} && ${DEPLOY_SCRIPT} --config ${deployConfigFile}`;
    await updatePackageScript(basePath, 'deploy', deployScript, fs);

    // undeploy script
    const undeployScript = `${buildPrefix}${UNDEPLOY_SCRIPT} --config ${deployConfigFile}`;
    await updatePackageScript(basePath, 'undeploy', undeployScript, fs);

    // test mode script
    const deployTestModeScript = `${BUILD_SCRIPT} && ${DEPLOY_SCRIPT} --config ${deployConfigFile} --testMode true`;
    await updatePackageScript(basePath, 'deploy-test', deployTestModeScript, fs);

    // dependencies
    await addPackageDevDependency(basePath, RIMRAF, RIMRAF_VERSION, fs);
}
