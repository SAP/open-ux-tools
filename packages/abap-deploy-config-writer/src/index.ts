import { join } from 'path';
import { create as createStorage } from 'mem-fs';
import { create } from 'mem-fs-editor';
import cloneDeep from 'lodash/cloneDeep';
import { addPackageDevDependency, FileName, readUi5Yaml, updatePackageScript } from '@sap-ux/project-access';
import { getDeployConfig, updateBaseConfig } from './config';
import { addUi5Dependency, writeUi5RepositoryFiles } from './file';
import {
    BUILD_SCRIPT,
    DEPLOY_SCRIPT,
    RIMRAF,
    RIMRAF_VERSION,
    UI5_TASK_FLATTEN_LIB,
    UI5_TASK_FLATTEN_LIB_VERSION,
    UNDEPLOY_SCRIPT
} from './constants';
import type { DeployConfigOptions } from './types';
import type { Editor } from 'mem-fs-editor';
import type { AbapDeployConfig } from '@sap-ux/ui5-config';

/**
 * Writes the template to the memfs editor instance.
 *
 * @param basePath - the base path
 * @param abapDeployConfig - the abap deploy config
 * @param options - the deploy config options
 * @param fs - the memfs editor instance
 * @returns the updated memfs editor instance
 */
async function generate(
    basePath: string,
    abapDeployConfig: AbapDeployConfig,
    options?: DeployConfigOptions,
    fs?: Editor
): Promise<Editor> {
    const abapConfig = cloneDeep(abapDeployConfig);

    if (!fs) {
        fs = create(createStorage());
    }

    // base config
    const baseConfigFile = options?.baseFile ?? FileName.Ui5Yaml;
    const baseConfig = await readUi5Yaml(basePath, baseConfigFile, fs);
    const baseConfigPath = join(basePath, baseConfigFile);
    updateBaseConfig(baseConfigPath, baseConfig, fs);

    // deploy config
    const deployConfigFile = options?.deployFile ?? FileName.UI5DeployYaml;
    const deployFilePath = join(basePath, deployConfigFile);
    const deployConfig = await getDeployConfig(abapConfig, baseConfig);
    fs.write(deployFilePath, deployConfig.toString());

    // package.json
    // deploy script
    const deployScript = `${BUILD_SCRIPT} && ${DEPLOY_SCRIPT} --config ${deployConfigFile} && ${RIMRAF} archive.zip`;
    await updatePackageScript(basePath, 'deploy', deployScript, fs);
    // undeploy script
    const undeployScript = `${BUILD_SCRIPT} && ${UNDEPLOY_SCRIPT} --config ${deployConfigFile}`;
    await updatePackageScript(basePath, 'undeploy', undeployScript, fs);
    // test mode script
    const deployTestModeScript = `${BUILD_SCRIPT} && ${DEPLOY_SCRIPT} --config ${deployConfigFile} --testMode true`;
    await updatePackageScript(basePath, 'deploy-test', deployTestModeScript, fs);
    // dependencies
    await addPackageDevDependency(basePath, RIMRAF, RIMRAF_VERSION, fs);
    await addPackageDevDependency(basePath, UI5_TASK_FLATTEN_LIB, UI5_TASK_FLATTEN_LIB_VERSION, fs);
    addUi5Dependency(fs, basePath, UI5_TASK_FLATTEN_LIB);

    // ui5 repo files
    await writeUi5RepositoryFiles(fs, basePath);

    return fs;
}

export { generate };
