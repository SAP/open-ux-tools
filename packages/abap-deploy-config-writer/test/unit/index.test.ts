import { join } from 'path';
import { generate } from '../../src';
import fsExtra from 'fs-extra';
import type { AbapDeployConfig, BspApp } from '@sap-ux/ui5-config';
import type { DeployConfigOptions } from '../../src/types';

describe('generate', () => {
    const outputDir = join(__dirname, '../test-output');
    debugger;
    const debug = true;

    beforeAll(async () => {
        fsExtra.removeSync(outputDir);
    });

    const config: AbapDeployConfig = {
        target: { url: 'https://example.com', client: '000', scp: false },
        app: { name: '', package: 'TESTPKG12', description: 'Mock deployment description', transport: 'TR123' }
    };

    const testConfigs: Array<{
        name: string;
        config: AbapDeployConfig;
        options?: DeployConfigOptions;
        isAppStudio: boolean;
    }> = [
        {
            name: 'test-js-app',
            config: {
                ...config,
                target: {
                    ...config.target,
                    authenticationType: 'reentranceTicket'
                }
            },
            isAppStudio: false
        },
        {
            name: 'test-ts-app',
            config: {
                ...config,
                target: {
                    ...config.target,
                    destination: 'MOCK_DESTINATION',
                    scp: true
                }
            },
            isAppStudio: true
        },
        {
            name: 'test.ui5.lib',
            config,
            options: {
                baseFile: 'base-config.yaml',
                deployFile: 'deploy-config.yaml'
            },
            isAppStudio: false
        },
        {
            name: 'test.ui5.typescript.library1',
            config,
            isAppStudio: false
        }
    ];

    test.each(testConfigs)('Generate deployment configs: $name', async ({ name, options, config, isAppStudio }) => {
        (config.app as BspApp).name = name;
        const testPath = join(outputDir, name);
        fsExtra.mkdirSync(outputDir, { recursive: true });
        fsExtra.mkdirSync(testPath);
        fsExtra.copySync(join(__dirname, `../sample/${name}`), testPath);

        const fs = await generate(testPath, config, options);
        expect(fs.dump(testPath)).toMatchSnapshot();

        return new Promise((resolve) => {
            if (debug) {
                debugger;
                fs.commit(resolve);
            } else {
                fsExtra.removeSync(testPath);
                resolve(true);
            }
        });
    });
});
