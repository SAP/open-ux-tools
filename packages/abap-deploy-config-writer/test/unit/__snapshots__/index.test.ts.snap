// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate Generate deployment configs: test.ui5.lib 1`] = `
Object {
  "base-config.yaml": Object {
    "contents": "specVersion: '2.2'
metadata:
  name: 'test.ui5.lib'
type: library
framework:
  name: SAPUI5
  version: 1.102.19
  libraries:
    - name: sap.ui.core
    - name: themelib_sap_belize
    - name: themelib_sap_fiori_3
server:
  customMiddleware:
    - name: fiori-tools-appreload
      afterMiddleware: compression
      configuration:
        path: test
        port: 35729
        delay: 300
builder:
  customTasks:
    - name: ui5-task-flatten-library
      afterTask: generateResourcesJson
",
    "state": "modified",
  },
  "deploy-config.yaml": Object {
    "contents": "# yaml-language-server: $schema=https://sap.github.io/ui5-tooling/schema/ui5.yaml.json

specVersion: '2.2'
metadata:
  name: 'test.ui5.lib'
type: library
framework:
  name: SAPUI5
  version: 1.102.19
  libraries:
    - name: sap.ui.core
    - name: themelib_sap_belize
    - name: themelib_sap_fiori_3
builder:
  resources:
    excludes:
      - /test/**
      - /localService/**
  customTasks:
    - name: deploy-to-abap
      afterTask: generateCachebusterInfo
      configuration:
        target:
          url: https://example.com
          client: '000'
        app:
          name: test.ui5.lib
          package: TESTPKG12
          description: Mock deployment description
          transport: TR123
        exclude:
          - /test/
",
    "state": "modified",
  },
  "package.json": Object {
    "contents": "{
  \\"name\\": \\"test.ui5.lib\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"devDependencies\\": {
    \\"@ui5/cli\\": \\"2.9.9\\",
    \\"rimraf\\": \\"^5.0.5\\",
    \\"ui5-task-flatten-library\\": \\"3.0.2\\"
  },
  \\"scripts\\": {
    \\"deploy\\": \\"npm run build && fiori deploy --config deploy-config.yaml && rimraf archive.zip\\",
    \\"undeploy\\": \\"npm run build && fiori undeploy --config deploy-config.yaml\\",
    \\"deploy-test\\": \\"npm run build && fiori deploy --config deploy-config.yaml --testMode true\\"
  },
  \\"ui5\\": {
    \\"dependencies\\": [
      \\"ui5-task-flatten-library\\"
    ]
  }
}
",
    "state": "modified",
  },
}
`;

exports[`generate Generate deployment configs: test-js-app 1`] = `
Object {
  "package.json": Object {
    "contents": "{
    \\"name\\": \\"test-js-app\\",
    \\"version\\": \\"0.0.1\\",
    \\"sapux\\": true,
    \\"description\\": \\"An SAP Fiori application.\\",
    \\"keywords\\": [
        \\"ui5\\",
        \\"openui5\\",
        \\"sapui5\\"
    ],
    \\"main\\": \\"webapp/index.html\\",
    \\"scripts\\": {
        \\"start\\": \\"start script\\",
        \\"build\\": \\"build script\\",
        \\"test\\": \\"test script\\",
        \\"deploy\\": \\"npm run build && fiori deploy --config ui5-deploy.yaml && rimraf archive.zip\\",
        \\"undeploy\\": \\"npm run build && fiori undeploy --config ui5-deploy.yaml\\",
        \\"deploy-test\\": \\"npm run build && fiori deploy --config ui5-deploy.yaml --testMode true\\"
    },
    \\"devDependencies\\": {
        \\"rimraf\\": \\"^5.0.5\\"
    }
}
",
    "state": "modified",
  },
  "ui5-deploy.yaml": Object {
    "contents": "# yaml-language-server: $schema=https://sap.github.io/ui5-tooling/schema/ui5.yaml.json

specVersion: '2.4'
metadata:
  name: 'test-js-app'
type: application
builder:
  resources:
    excludes:
      - /test/**
      - /localService/**
  customTasks:
    - name: deploy-to-abap
      afterTask: generateCachebusterInfo
      configuration:
        target:
          url: https://example.com
          client: '000'
          authenticationType: reentranceTicket
        app:
          name: test-js-app
          package: TESTPKG12
          description: Mock deployment description
          transport: TR123
        exclude:
          - /test/
",
    "state": "modified",
  },
}
`;

exports[`generate Generate deployment configs: test-ts-app 1`] = `
Object {
  "package.json": Object {
    "contents": "{
    \\"name\\": \\"test-ts-app\\",
    \\"version\\": \\"0.0.1\\",
    \\"private\\": true,
    \\"description\\": \\"An SAP Fiori application.\\",
    \\"keywords\\": [
        \\"ui5\\",
        \\"openui5\\",
        \\"sapui5\\"
    ],
    \\"main\\": \\"webapp/index.html\\",
    \\"dependencies\\": {},
    \\"devDependencies\\": {
        \\"rimraf\\": \\"^5.0.5\\"
    },
    \\"scripts\\": {
        \\"start\\": \\"start script\\",
        \\"build\\": \\"build script\\",
        \\"test\\": \\"test script\\",
        \\"deploy\\": \\"npm run build && fiori deploy --config ui5-deploy.yaml && rimraf archive.zip\\",
        \\"undeploy\\": \\"npm run build && fiori undeploy --config ui5-deploy.yaml\\",
        \\"deploy-test\\": \\"npm run build && fiori deploy --config ui5-deploy.yaml --testMode true\\"
    },
    \\"sapuxLayer\\": \\"VENDOR\\"
}
",
    "state": "modified",
  },
  "ui5-deploy.yaml": Object {
    "contents": "# yaml-language-server: $schema=https://sap.github.io/ui5-tooling/schema/ui5.yaml.json

specVersion: '3.1'
metadata:
  name: test-ts-app
type: application
builder:
  customTasks:
    - name: ui5-tooling-transpile-task
      afterTask: replaceVersion
      configuration:
        debug: true
        transformModulesToUI5:
          overridesToOverride: true
    - name: deploy-to-abap
      afterTask: generateCachebusterInfo
      configuration:
        target:
          destination: MOCK_DESTINATION
          url: https://example.com
          client: '000'
          scp: true
        app:
          name: test-ts-app
          package: TESTPKG12
          description: Mock deployment description
          transport: TR123
        exclude:
          - /test/
  resources:
    excludes:
      - /test/**
      - /localService/**
",
    "state": "modified",
  },
  "webapp/.Ui5RepositoryTextFiles": Object {
    "contents": "^.*.ts$",
    "state": "modified",
  },
}
`;
